src/EthereumHelpers.js

```diff
@@ -233,29 +235,34 @@ async function sendSafelyRetryable(
   forceSend,
   totalAttempts
 ) {
-  for (let attempt = 1; true; attempt++) {
-    try {
-      console.debug(`sending transaction; attempt number ${attempt}`)
-
-      return await sendSafely(boundContractMethod, sendParams, forceSend)
-    } catch (exception) {
-      if (attempt === totalAttempts) {
-        console.debug(`last attempt ${attempt} failed; throwing exception`)
-        throw exception
-      }
-
-      const backoffMillis = Math.pow(2, attempt) * 1000
-      const jitterMillis = Math.floor(Math.random() * 100)
-      const waitMillis = backoffMillis + jitterMillis
-
-      console.debug(
-        `attempt ${attempt} failed: ${exception}; ` +
-          `retrying after ${waitMillis} milliseconds`
-      )
+  return backoffRetrier(totalAttempts)(async () => {
+    return await sendSafely(boundContractMethod, sendParams, forceSend)
+  })
+}
 
-      await new Promise(resolve => setTimeout(resolve, waitMillis))
-    }
-  }
+/**
+ * Calls the bound contract method (using `.call`, that is as a read-only call)
+ * and retries up to `totalAttempts` number of times, or 3 if unspecified.
+ *
+ * @param {ContractSendMethod} boundContractMethod A bound web3 contract method
+ *        with `estimateGas`, `send`, and `call` variants available.
+ * @param {Partial<SendOptions>} [sendParams] The parameters to pass to `call`.
+ * @param {number} [totalAttempts=3] Total attempts which should be performed in
+ *        case of an error before rethrowing it to the caller.
+ *
+ * @return {Promise<any>} A promise to the result of sending the bound contract
+ *         method. Fails the promise if gas estimation fails, extracting an
+ *         on-chain error if possible.
+ */
+async function callWithRetry(boundContractMethod, sendParams, totalAttempts) {
+  const retries = typeof totalAttempts === "undefined" ? 3 : totalAttempts
```

Comment:
Could this line be skipped if we just use a default param value for `totalAttempts` like:
```
async function callWithRetry(boundContractMethod, sendParams, totalAttempts = 3) {
```

Reply:
I need to see when node started supporting that... I think it's too new?

Reply:
(Target being Node 12.)

Reply:
Ah right, good point.

Reply:
Confirmed this works :thumbsup: